### PR TITLE
Switch to stretch based image and include systemd by default.

### PR DESCRIPTION
**Pull request by bmoyles0117**

### CODE REVIEW

**Comment by qingling128** [comment]
Mind providing some context around why we are making this switch?

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,19 @@
-FROM ubuntu:xenial
+FROM debian:stretch
 
-RUN apt-get update \
+RUN sed -i 's/101/0/' /usr/sbin/policy-rc.d \
```
Comment (qingling128) [inline docker/Dockerfile]: Maybe add a comment describing why we need to replace `101` with `0`? It's a bit confusing just looking at it...

File: docker/run.sh [hunk 2]
```diff
@@ -1,14 +1,6 @@
 # For systems without journald.
 mkdir -p /var/log/journal
 
-# Copy host libsystemd into image to avoid compatibility issues. This operation
```
Comment (qingling128) [inline docker/run.sh]: Is this the reason why we are making the switch?

**Comment by igorpeshansky** [comment]
@qingling128 See https://github.com/GoogleCloudPlatform/k8s-stackdriver/pull/101.

File: docker/run.sh (hunk 2 above)
Comment (bmoyles0117) [reply docker/run.sh]: Yes, these initial additions as well as current deletions are based on findings from the fluentd-gcp image here: https://github.com/GoogleCloudPlatform/k8s-stackdriver/pull/101

Comment (qingling128) [reply docker/run.sh]: Cool cool. Thanks for the clarification!

File: docker/Dockerfile (hunk 1 above)
Comment (bmoyles0117) [reply docker/Dockerfile]: Done.

**Comment by bmoyles0117** [comment]
PTAL

File: docker/Dockerfile [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
+        gnupg2 \
         lsb-release \
+        procps \
+        systemd \
     && curl -sS https://dl.google.com/cloudagents/install-logging-agent.sh | bash \
     && apt-get autoremove \
     && sed -i "s/num_threads 8/num_threads 8\n  detect_json true\n  # Enable metadata agent lookups.\n  enable_metadata_agent true\n  metadata_agent_url \"http:\/\/local-metadata-agent.stackdriver.com:8000\"/" "/etc/google-fluentd/google-fluentd.conf" \
-    && rm -rf /var/lib/apt/lists/*
+    && rm -rf /var/lib/apt/lists/* \
+    && rm /tmp/* \
```
Comment (igorpeshansky) [inline docker/Dockerfile]: `rm -rf`, just in case directories get created under `/tmp`?

Comment (bmoyles0117) [reply docker/Dockerfile]: Done.

**Comment by qingling128** [comment]
LGTM.

**Pull request merged**